docs/source/intro.dox

```diff
@@ -365,7 +368,369 @@ unit tests with `make tests` and run
+
+For our custom operator, let's add two scratchpads to the Reference backend.
+Open up backend.cpp and add two array definitions as shown below:
+
+\code
+namespace ref {
+// This is all existing code...
+const unsigned kConvolutionHw = 0x0001;
+const unsigned kInnerProductHw = 0x0002;
+const unsigned kEltwiseOpHw = 0x0003;
+const unsigned kBatchNormHw = 0x0004;
+const unsigned kPoolingHw = 0x0005;
+
+// Define our new scratchpads here.
+int kSpadSize;
+float* spad0;
+float* spad1;
+
+// Add a unique ID for our accelerator HW. This will be used to invoke the
+// accelerator during simulation.
+const unsigned kMyCustomOperatorHw = 0x00006;
+}  // namespace ref
+\endcode
+
+Then, open up backend.h and add extern declarations for them:
+
+\code
+namespace ref {
+// This is all existing code...
+extern const unsigned kConvolutionHw;
+extern const unsigned kInnerProductHw;
+extern const unsigned kEltwiseOpHw;
+extern const unsigned kBatchNormHw;
+extern const unsigned kPoolingHw;
+
+// Declare our two new global arrays and accelerator IDs here.
+extern int kSpadSize;
+extern float* spad0;
+extern float* spad1;
+extern const unsigned kMyCustomOperatorHw;
+}  // namespace ref
+\endcode
+
+We need to allocate memory for these new arrays before they can be used. By
+this point, you should have picked a scratchpad size. Let's say you picked
+32KB. In backend.h, modify `ReferenceBackend` like so:
+
+\code
+class ReferenceBackend {
+  static int SpadSize() { return ref::kSpadSize; }
+  static void initGlobals() {
+    ref::kSpadSize = 32*1024;  // Replace with your actual value.
+    ref::spad0 = (float*) malloc_aligned(ref::kSpadSize);
+    ref::spad1 = (float*) malloc_aligned(ref::kSpadSize);
+  }
+  static void freeGlobals() {
+    free(ref::spad0);
+    free(ref::spad1);
+  }
+}
+\endcode
+
+Your accelerator-local scratchpads have now been correctly configured. We will
+modify our kernel function to use these shortly.
+
+\subsubsection tiling Tile your Tensors
+
+Writing logic to correctly tile your tensors can be tricky. There are lots of
+corner cases and special features of the data that must be taken into account
+so that the accelerator can still compute the correct output with partial data.
+To ease this, SMAUG provides a library of useful tiling functionality that you
+can reuse. When that library is insufficient, you can refer to the existing
+tiling optimizers (e.g. for SMV convolution) to get started on more complex
+tiling scenarios.
+
+Fortunately, tiling a tensor for an elementwise operator is the easiest kind of
+tiling there is - no need to consider data reuse, striding, etc. We simply need
+to break up the two input tensors and output tensor into equal sized pieces
+which maximize the use of the local scratchpads. This can be accomplished with
+just a few lines of code in `my_custom_operator.h` by overriding the `tile()`
+function and calling the smaug::generateTiledTensorSimple function. The result
```

Comment:
I think this should be `generateTiledTensorPerBatchNC`?

Reply:
Yes, good catch. Done.